packages/design-editor/test/tokens/css-decls.test.ts

```
@@ -0,0 +1,370 @@
+
+  it('does NOT see a block nested inside an at-rule — a documented limit', () => {
+    // A `@media (prefers-color-scheme: dark)` project reads as having no dark theme
+    // rather than as having a wrong one. Asserted so the limit is a decision on record,
+    // not an accident that a later change can quietly reverse.
+    const css = `@media (prefers-color-scheme: dark) {\n  :root { --a: 1px; }\n}\n`;
+    expect(readDecls(css, ':root')).toEqual([]);
+  });
+
+  it('returns duplicates rather than merging them', () => {
+    const css = ':root { --a: 1px; }\n:root { --a: 5px; }\n';
+    expect(readDecls(css, '--a' === '--a' ? ':root' : ':root').map((d) => d.value)).toEqual([
```

Biome (2.5.6)

[error] This comparison uses the same expression on both sides.

(lint/suspicious/noSelfCompare)